src/app/clusters/energy-preference-server/energy-preference-server.cpp

```diff
@@ -0,0 +1,217 @@
+        {
+            return aEncoder.EncodeList([endpoint](const auto & encoder) -> CHIP_ERROR {
+                BalanceStruct::Type balance;
+                size_t index   = 0;
+                CHIP_ERROR err = CHIP_NO_ERROR;
+                while ((err = gsDelegate->GetEnergyBalanceAtIndex(endpoint, index, balance)) == CHIP_NO_ERROR)
+                {
+                    ReturnErrorOnFailure(encoder.Encode(balance));
+                    index++;
+                }
+                if (err == CHIP_ERROR_NOT_FOUND)
+                {
+                    return CHIP_NO_ERROR;
+                }
+                return err;
+            });
+        }
+        return CHIP_ERROR_INCORRECT_STATE;
+    case EnergyPriorities::Id:
+        if (balanceSupported == false)
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+
+        if (gsDelegate != nullptr)
+        {
+            return aEncoder.EncodeList([endpoint](const auto & encoder) -> CHIP_ERROR {
+                EnergyPriorityEnum priority;
+                size_t index   = 0;
+                CHIP_ERROR err = CHIP_NO_ERROR;
+                while ((err = gsDelegate->GetEnergyPriorityAtIndex(endpoint, index, priority)) == CHIP_NO_ERROR)
+                {
+                    ReturnErrorOnFailure(encoder.Encode(priority));
+                    index++;
+                }
+                if (err == CHIP_ERROR_NOT_FOUND)
+                {
+                    return CHIP_NO_ERROR;
+                }
+                return err;
+            });
+        }
+        return CHIP_ERROR_INCORRECT_STATE;
+    case LowPowerModeSensitivities::Id:
+        if (lowPowerSupported == false)
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+
+        if (gsDelegate != nullptr)
+        {
+            return aEncoder.EncodeList([endpoint](const auto & encoder) -> CHIP_ERROR {
+                BalanceStruct::Type balance;
+                size_t index   = 0;
+                CHIP_ERROR err = CHIP_NO_ERROR;
+                while ((err = gsDelegate->GetLowPowerModeSensitivityAtIndex(endpoint, index, balance)) == CHIP_NO_ERROR)
+                {
+                    ReturnErrorOnFailure(encoder.Encode(balance));
+                    index++;
+                }
+                if (err == CHIP_ERROR_NOT_FOUND)
+                {
+                    return CHIP_NO_ERROR;
+                }
+                return err;
+            });
+        }
+        return CHIP_ERROR_INCORRECT_STATE;
+    default: // return CHIP_NO_ERROR and just read from the attribute store in default
+        break;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+} // anonymous namespace
+
+namespace chip::app::Clusters::EnergyPreference
+{
+
+void SetDelegate(Delegate * aDelegate)
+{
+    gsDelegate = aDelegate;
+}
+
+Delegate * GetDelegate()
+{
+    return gsDelegate;
+}
+
+} // Set matter energy preferences delegate
+
+Protocols::InteractionModel::Status
```

Comment:
```suggestion
Status
```
Why have an `using` if you're not going to use the results?